src/ivc/step_folding_circuit.rs

```diff
@@ -233,44 +233,30 @@ where
         config: StepConfig<ARITY, C::Base, SC, T>,
         mut layouter: impl Layouter<C::Base>,
     ) -> Result<(), halo2_proofs::plonk::Error> {
-        let assigned_z_i: [_; ARITY] = layouter.assign_region(
-            || "assigned_zi",
-            |region| {
-                let mut region = RegionCtx::new(region, 0);
-
-                config
-                    .main_gate_config
-                    .advice_cycle_assigner()
-                    .assign_all_advice(&mut region, || "z0_primary", self.input.z_0.iter().copied())
-                    .map(|inp| inp.try_into().unwrap())
-            },
-        )?;
-        debug!("assigned z_i");
-
-        let assigned_z_0: [_; ARITY] = layouter
+        let (assigned_z_0, assigned_z_i): ([_; ARITY], [_; ARITY]) = layouter
             .assign_region(
-                || "assigned_z0_primary",
+                || "assign z_0 & z_i",
                 |region| {
                     let mut region = RegionCtx::new(region, 0);
 
-                    config
-                        .main_gate_config
-                        .advice_cycle_assigner()
-                        .assign_all_advice(
-                            &mut region,
-                            || "z0_primary",
-                            self.input.z_0.iter().copied(),
-                        )
-                        .map(|inp| inp.try_into().unwrap())
+                    let mut assigner = config.main_gate_config.advice_cycle_assigner();
+
+                    let z_0 = assigner
+                        .assign_all_advice(&mut region, || "z_0", self.input.z_0.iter().copied())
+                        .map(|inp| inp.try_into().unwrap())?;
+
+                    let z_i = assigner
```

Comment:
When call assign_all_advice twice, will the second call continue the assignment from the end column location of the first call?